joplin/base/models/janis_url.py

```diff
@@ -0,0 +1,199 @@
+
+
+class ServicePageJanisUrl(ClusterableModel):
+    page = ParentalKey(ServicePage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class InformationPageJanisUrl(ClusterableModel):
+    page = ParentalKey(InformationPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class LocationPageJanisUrl(ClusterableModel):
+    page = ParentalKey(LocationPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class DepartmentPageJanisUrl(ClusterableModel):
+    page = ParentalKey(DepartmentPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class TopicCollectionPageJanisUrl(ClusterableModel):
+    page = ParentalKey(TopicCollectionPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class TopicPageJanisUrl(ClusterableModel):
+    page = ParentalKey(TopicPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+# @register_snippet
+class JanisUrl(models.Model):
+    url = models.CharField(max_length=9001)
+    language = models.CharField(max_length=9)
+
+    # Themes don't have dedicated pages
+    theme = models.ForeignKey('base.Theme', on_delete=models.PROTECT,null=True, blank=True)
+
+    topic_collection_page = models.ForeignKey('base.TopicCollectionPage', on_delete=models.CASCADE, null=True, blank=True)
+    topic_page = models.ForeignKey('base.TopicPage', on_delete=models.CASCADE, null=True, blank=True)
+    department_page = models.ForeignKey("base.departmentPage",on_delete=models.PROTECT, null=True, blank=True)
+    location_page = models.ForeignKey("locations.LocationPage",on_delete=models.PROTECT, null=True, blank=True)
+    information_page = models.ForeignKey("base.InformationPage", on_delete=models.PROTECT, null=True, blank=True)
+    service_page = models.ForeignKey("base.ServicePage", on_delete=models.PROTECT, null=True, blank=True)
+    guide_page = models.ForeignKey("base.GuidePage", on_delete=models.PROTECT, null=True, blank=True)
+    official_documents_page = models.ForeignKey("base.OfficialDocumentPage", on_delete=models.PROTECT, null=True, blank=True)
+    form_container = models.ForeignKey("base.FormContainer", on_delete=models.PROTECT, null=True, blank=True)
+
+    def __str__(self):
+        return self.url
+
+    @staticmethod
+    def get_translated_slug(language, page):
+        # todo: flesh this out
+        if language == 'es':
```

Comment:
Is there a reason why we can't use this pattern? 
https://docs.wagtail.io/en/v2.7.1/advanced_topics/i18n/index.html#serving-different-languages-from-different-urls

Reply:
In my understanding that pattern is about actual URLs, where as these aren't actually urls but instead just strings that represent what the URL will be after a Janis build.

tl;dr - not sure how we'd be able to do that, any suggestions are very welcome